### PR TITLE
ddl : ddl forbid partition table switch into cache and add view feature test

### PR DESCRIPTION
**Pull request by JayL-zxl**
### What problem does this PR solve?

Issue Number: https://github.com/pingcap/tidb/issues/29318

Problem Summary:

This is a subtask of https://github.com/pingcap/tidb/issues/25293
### What is changed and how it works?
+ Forbid partition table switch cacheable`alter partition table cache` is not supported
+ Add view feature test, `alter view cache`is invalid.
+ Created a file specially used to test the operation of the cache table ddl,  and integrated some tests into it.
### Check List 

Tests 

-  Unit test

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by JayL-zxl** [comment]
./rebuild

**Comment by JayL-zxl** [comment]
@djshow832  @tisonkun PTAL thks

**Review by lcwangchao — APPROVED** [review]

File: ddl/db_cache_test.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
+// Copyright 2018 PingCAP, Inc.
```
[the remark this answers is not in the document]
Comment (tiancaiamao) [reply ddl/db_cache_test.go:1]: Rest LGTM

[the remark this answers is not in the document]
Comment (JayL-zxl) [reply ddl/db_cache_test.go]: sorry. my mistake

**Review by tiancaiamao — APPROVED** [review]

**Pull request merged**